### PR TITLE
Cylinder type 2D/3D

### PR DESCRIPTION
**Pull request by edljk**
New try with more flexibility. As previously feel free to modify..
Any comments/suggestions of improvement are welcome !
The following example illustrates the 2D/3D possibilities

```julia
using GLVisualize, GeometryTypes, Colors

function hairyball(dimc::Int64=rand(2:3,1)[1])
  large_sphere = HyperSphere(Point{dimc,Float32}(0),1f0)
  positions = GeometryTypes.decompose(Point{dimc,Float32},large_sphere); nbc = length(positions)
  cc = Array{GeometryTypes.Cylinder{dimc,Float32},1}(nbc)
  
  window = glscreen()
  if dimc==2
    for k = 1:length(positions)
      positions[k] = Point3f0([positions[k]...,0])
    end
  end
  spheres = visualize((HyperSphere{3,Float32}(Point3f0(0.0), 1f0),
  positions), scale = Vec3f0(0.05))
  _view(spheres,window)
  resolution = dimc==2 ? (2,3) : 80
  @async renderloop(window)
  for k = 1:nbc
    pp = positions[k]
    cc[k] = GeometryTypes.Cylinder{dimc,Float32}(pp,pp*1.2,Float32(0.05+0*rand()/10))
    vis = visualize(GLNormalMesh(cc[k],resolution),color=RGBA{Float32}(rand(),rand(),rand()))
    _view(vis,window)
  end
  @async renderloop(window)
end

hairyball()
```

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -179,6 +179,16 @@ immutable FlexibleSimplex{T} <: AFG{T}
 end
 
 """
+A `Cylinder` is a 2D rectangle or a 3D cylinder defined by its origin point,
+its extremity and a radius. `origin`, `extremity` and `r`, must be specified.
+"""
+immutable Cylinder{N,T<: AbstractFloat} <: GeometryPrimitive{N,T<: AbstractFloat}
```
Comment (SimonDanisch) [inline src/types.jl]: `Cylinder{N,T<: AbstractFloat} <: GeometryPrimitive{N,T}` should be enough (and won't error on 0.4)!
Also, AbstractFloat is maybe a bit too strict! Use Number?

File: src/cylinder.jl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+origin{N,T}(c::Cylinder{N,T}) = c.origin
+extremity{N,T}(c::Cylinder{N,T}) = c.extremity
+radius{N,T}(c::Cylinder{N,T}) = c.r
+height{N,T}(c::Cylinder{N,T}) = norm(c.extremity-c.origin)
+direction{N,T}(c::Cylinder{N,T}) = (c.extremity.-c.origin)./height(c)
+
+function rotation{T}(c::Cylinder{2,T})
+  u = [direction(c)...,0]; v = [u[2],-u[1],0]
```
Comment (SimonDanisch) [inline src/cylinder.jl]: Would be nice if you can also use 4 spaces! :)

File: src/types.jl (hunk 1 above)
Comment (edljk) [inline src/types.jl]: Actually the constraint  ` AbstractFloat`   comes from the fact that even a rotated rectangle with ` Int` coordinates may have no Integer coordinates.. I do not know what would be the best way  to fix this.

**Comment by SimonDanisch** [comment]
*sigh* win32...

File: src/types.jl (hunk 1 above)
Comment (SimonDanisch) [inline src/types.jl]: It's not very important ;) We can still fix it if someone comes along with a good reason why they want another number type!

**Comment by edljk** [comment]
I do not know what  to do to help for this win32 problem.. ??

**Comment by SimonDanisch** [comment]
Looks like `isapprox` can fix this!

**Comment by edljk** [comment]
Any additional suggestion (I have no idea of the origin of failing tests..) ?

**Comment by SimonDanisch** [comment]
I'm fixing this in #82

**Comment by SimonDanisch** [comment]
merged in: #82

**Pull request closed without merging**